app/assets/javascripts/filters.js

```diff
@@ -0,0 +1,148 @@
+    const categoryId = $isDefaultCheckbox.val();
+    var defaultFilter = await QPixel.defaultFilter(categoryId);
+    const $deleteButton = $form.find('.filter-delete');
+
+    // Enables/Disables Save & Delete buttons programatically
+    async function computeEnables() {
+      const filters = await QPixel.filters();
+      const filterName = $select.val();
+
+      // Nothing set
+      if (!filterName) {
+        $saveButton.prop('disabled', true);
+        $deleteButton.prop('disabled', true);
+        return;
+      }
+
+      const filter = filters[filterName]
+
+      // New filter
+      if (!filter) {
+        $saveButton.prop('disabled', false);
+        $deleteButton.prop('disabled', true);
+        return;
+      }
+
+      // Not a new filter
+      $deleteButton.prop('disabled', filter.system);
+
+      const hasChanges = [...$formFilters].some(el => {
+        const filterValue = filter[el.dataset.name];
+        let elValue = $(el).val();
+        if (filterValue?.constructor == Array) {
+          elValue = elValue ?? [];
+          return filterValue.length != elValue.length || filterValue.some((v, i) => v[1] != elValue[i]);
+        }
+        else {
+          return filterValue ? filterValue != elValue : elValue;
+        }
+      });
+      const defaultStatusChanged = $isDefaultCheckbox.prop('checked') != (defaultFilter === $select.val());
+      $saveButton.prop('disabled', !defaultStatusChanged && (filter.system || !hasChanges));
+    }
+
+    async function initializeSelect() {
+      defaultFilter = await QPixel.defaultFilter(categoryId);
+      $isDefaultCheckbox.prop('checked', defaultFilter === $select.val());
+      const filters = await QPixel.filters();
+
+      function template(option) {
+        if (option.id == '') { return 'None'; }
+
+        const filter = filters[option.id];
+        const name = `<span>${option.text}</span>`;
+        const systemIndicator = filter?.system
+          ? ' <span has-font-size-caption">(System)</span>'
+          : '';
+        const newIndicator = !filter
+          ? ' <span has-font-size-caption">(New)</span>'
+          : '';
+        return $(name + systemIndicator + newIndicator);
+      }
+
+      // Clear out any old options
+      $select.children().filter((_, option) => option.value && !filters[option.value]).detach();
+      $select.select2({
+        data: Object.keys(filters),
+        tags: true,
+
+        templateResult: template,
+        templateSelection: template
+      }).on('select2:select', async evt => {
+        const filterName = evt.params.data.id;
+        const preset = filters[filterName];
+
+        $isDefaultCheckbox.prop('checked', defaultFilter === $select.val());
+        computeEnables();
+
+        // Name is not one of the presets, i.e user is creating a new preset
+        if (!preset) {
+          return;
+        }
+
+        for (const [name, value] of Object.entries(preset)) {
```

Comment:
consider (es+, non-essential): `Object.entries(preset).forEach((k, v) => {` etc.

Reply:
My usage of `for...of` intead of `.forEach` is mostly personal preference since I think it's more readable. Is there a benefit to the `.forEach` version that I'm missing?

Reply:
Nope, likewise that's just preference, that's why it's a consider :)